modules/registry/registry.go

```diff
@@ -205,6 +204,52 @@ func (c *RegistryContainer) PushImage(ctx context.Context, ref string) error {
 	return c.ImageExists(ctx, ref)
 }
 
+// PullImage pulls an image from an external Registry.
+// It will use the internal registry to store the image.
+// This function is helpful when you want to push an image to your Registry
+// instance made by testcontainer.
+func (c *RegistryContainer) PullImage(ctx context.Context, ref string) error {
+	dockerCli, err := testcontainers.NewDockerClientWithOpts(ctx)
+	if err != nil {
+		return fmt.Errorf("create docker client: %w", err)
+	}
+	defer dockerCli.Close()
+
+	pullOpts := image.PullOptions{
+		All: true,
+	}
+
+	output, err := dockerCli.ImagePull(ctx, ref, pullOpts)
+	if err != nil {
+		return fmt.Errorf("failed to push image %s: %w", ref, err)
```

Comment:
suggestion: sorry to be a pain, I think you missed my mention of more below 😄 
I'll flag each one to make it easier
```suggestion
		return fmt.Errorf("pull image %q: %w", ref, err)
```